flink-ml-core/src/main/java/org/apache/flink/ml/linalg/BLAS.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.linalg;
+
+import org.apache.flink.util.Preconditions;
+
+/** A utility class that provides BLAS routines over matrices and vectors. */
+public class BLAS {
+
+    private static final dev.ludovic.netlib.BLAS NATIVE_BLAS =
+            dev.ludovic.netlib.BLAS.getInstance();
+
+    /**
+     * \sum_i |x_i| .
+     *
+     * @param x x
+     * @return
+     */
+    public static double asum(double[] x) {
+        return NATIVE_BLAS.dasum(x.length, x, 0, 1);
+    }
+
+    /**
+     * y += a * x .
+     *
+     * @param a a
+     * @param x x
+     * @param y y
+     */
+    public static void axpy(double a, double[] x, double[] y) {
```

Comment:
In the future will we need to support `axpy(a: Double, x: SparseVector, y: DenseVector)`? If so, should we make this `axpy(a: Double, x: DenseVector, y: DenseVector)`?

Same for the methods in BLAS.

Reply:
Thanks for adding `public static void axpy(double a, DenseVector x, DenseVector y)`.  Could we consistently use `DenseVector` instead of `double[]` in `BLAS.java`  to avoid having two public APIs for doing the same thing? And maybe apply this rule consistently for all methods in BLAS such as `dot(...)`.